hie-compat/hie-compat.cabal

```diff
@@ -24,8 +24,8 @@ library
   default-language:    Haskell2010
   build-depends:
      base < 4.17, array, bytestring, containers, directory, filepath, transformers
-  if flag(ghc-lib)
-    build-depends: ghc-lib
+  if flag(ghc-lib) && impl(ghc < 9)
```

Comment:
This (and other .hs related changes) was introduced after a rebase from one of @pepeiborra branch.